src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.cpp

```diff
@@ -127,13 +127,18 @@ void ShenandoahHeapRegion::make_young_maybe() {
    case _cset:
    case _humongous_start:
    case _humongous_cont:
+
+
+
      if (affiliation() != YOUNG_GENERATION) {
        ShenandoahHeap* heap = ShenandoahHeap::heap();
        if (heap->mode()->is_generational()) {
          if (is_old()) {
            heap->old_generation()->decrement_affiliated_region_count();
+           heap->old_generation()->decrease_capacity(region_size_bytes());
```

Comment:
Should capacity be updated every time affiliated region count changes? If yes, then we could have increase/decrease affiliated region count increase/decrease capacity directly. If no, is this something peculiar to full GC?

Reply:
This is a good observation and suggestion.  This doesn't seem to be a consistent behavior.  promote_in_place() adjusts affiliated regions without adjusting capacity, but maybe it should.  I'll explore further.

Reply:
Looks like this is something special about full gc.  The call to make_young_maybe() only occurs when !_is_generational().  kind of a peculiar name for this function, given that we're non-generational here.  Still exploring.

Reply:
That code should not even be reached.  I'm testing a change here.

Reply:
Have replace this code with an assert.  Thanks for catching the inconsistency.